######################################################################
#################### My Project 3 RNAseq Pipeline ####################
######################################################################

# Read in the data
filepath = '/projectnb/bf528/materials/project_3_individual/rnaseq'
#filepath = '/projectnb/bf528/students/nmf35/bf528-individual-project-NathanielFisher1/test_samples'
# Name reps
reps = ['rep1','rep2','rep3']
#reps = ['rep1','rep2']

# Name samples
names = ['CTL', 'KO']
#names = ['sub_CTL', 'sub_KO']

# Name reads
reads = ['R1','R2']

# other inputs
#                 'results/multiqc_report.html'


# First perform QC on reads using fastqc and assemble everything into one report with multiqc
rule all:
		input:
				'results/verse_concat_filtered.csv',
				'results/id2gene.txt'
				

# Run fastqc for each sample to check read quality
rule fastqc:
		input:
				fastq = expand('/projectnb/bf528/materials/project_3_individual/rnaseq/{name}{rep}_{read}.fastq.gz', name = names, rep = reps, read = reads)
		output:
				'results/{name}{rep}_{read}_fastqc.html'
		params:
				outdir = 'results/'
		threads: 4 
		conda:
				'envs/fastqc_env.yml'
		shell:
				'''
				fastqc {input.fastq} --outdir {params.outdir}
				'''

# Perform read alignment for each sample

# First need to get human genome

rule wget_hg38_fa:
		output:
				'results/GRCh38.primary_assembly.genome.fa.gz'
		params:
				ftp_path = 'https://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_45/GRCh38.primary_assembly.genome.fa.gz'
		shell:
				'''
				wget {params.ftp_path} -P ./results/
				'''

# Then need to get human gtf

rule wget_hg38_gtf:
		output:
				'results/gencode.v45.primary_assembly.annotation.gtf.gz'
		params:
		shell:
				'''
				wget https://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_45/gencode.v45.primary_assembly.annotation.gtf.gz -P ./results/
				'''

rule unzip_gtf:
		input:
				gz = 'results/gencode.v45.primary_assembly.annotation.gtf.gz'
		output:
				gtf = 'results/gencode.v45.primary_assembly.annotation.gtf'
		shell:
				'''
				gunzip -c {input.gz} > {output.gtf}
       			'''

rule unzip_fa:
		input:
				gz = 'results/GRCh38.primary_assembly.genome.fa.gz'
		output:
				fa = 'results/GRCh38.primary_assembly.genome.fa'
		shell:
				'''
				gunzip -c {input.gz} > {output.fa}
				'''

# Now align our reads using STAR (using --outSAMunmapped Within so that I can look at alignment quality with samtools flagstats )

rule star_index:
		input:
				fa = 'results/GRCh38.primary_assembly.genome.fa',
				star_dir = 'samples/star_dir',
				gtf = 'results/gencode.v45.primary_assembly.annotation.gtf'
		output:
				'samples/star_dir/chrLength.txt',
				'samples/star_dir/chrNameLength.txt',
				'samples/star_dir/chrName.txt',
				'samples/star_dir/chrStart.txt',
				'samples/star_dir/exonGeTrInfo.tab',
				'samples/star_dir/exonInfo.tab',
				'samples/star_dir/geneInfo.tab',
				'samples/star_dir/Genome',
				'samples/star_dir/genomeParameters.txt',
				'samples/star_dir/SA',
				'samples/star_dir/SAindex',
				'samples/star_dir/sjdbInfo.txt',
				'samples/star_dir/sjdbList.fromGTF.out.tab',
				'samples/star_dir/sjdbList.out.tab',
				'samples/star_dir/transcriptInfo.tab'
		params:
		conda:
				'envs/star_env.yml'
		threads: 4
		shell:
				'''
				STAR --runThreadN {threads} --runMode genomeGenerate --genomeDir {input.star_dir} --genomeFastaFiles {input.fa} \
				--sjdbGTFfile {input.gtf}
				'''



rule star:
		input:
				'samples/star_dir/transcriptInfo.tab',     #make star dir output an input here
				r1 = '/projectnb/bf528/materials/project_3_individual/rnaseq/{name}{rep}_R1.fastq.gz',
				r2 = '/projectnb/bf528/materials/project_3_individual/rnaseq/{name}{rep}_R2.fastq.gz',
				star_dir = 'samples/star_dir/'
		output:	
				bam = 'results/{name}{rep}.Aligned.out.bam'
		params:
				prefix = 'results/{name}{rep}.'
		conda:
				'envs/star_env.yml'
		shell:
				'''
				STAR --genomeDir {input.star_dir} --readFilesIn {input.r1} {input.r2} --readFilesCommand zcat --outSAMtype BAM Unsorted --outFileNamePrefix {params.prefix} --outSAMunmapped Within
				'''

# Perform alignment QC with samtools flagstats

rule samtools_flagstat:
		input:
				bam = 'results/{name}{rep}.Aligned.out.bam'
		output:
				flagstats = 'results/{name}{rep}_flagstats.txt'
		conda:
				'envs/samtools_env.yml'
		shell:
				'samtools flagstats {input.bam} > {output.flagstats}'

# remember that you want multiqc to run only after fastqc has run on all the files
rule multiqc:
		input:
				expand('results/{name}{rep}_{read}_fastqc.html', path = filepath, name = names, rep = reps, read = reads),
				expand('results/{name}{rep}_flagstats.txt', path = filepath, name = names, rep = reps)
		output:
				'results/multiqc_report.html'
		conda:
				'envs/multiqc_env.yml'
		shell:
				'''
				multiqc . -o results/
				'''

rule verse:
		input:
				'results/multiqc_report.html',   #run multiqc before verse
				bam_file = 'results/{name}{rep}.Aligned.out.bam',
				gtf_file = 'results/gencode.v45.primary_assembly.annotation.gtf'
		output:
				exons_file = 'results/{name}{rep}.exon.txt'
		params:
				prefix = 'results/{name}{rep}'
		conda:
				'envs/verse_env.yml'
		shell:
				'''
				    verse -S -a {input.gtf_file} -o {params.prefix} {input.bam_file}
				'''

rule concat_verse:
		input:
				exon_files = expand('results/{name}{rep}.exon.txt', name=names, rep = reps)
		output:
				all_exon_files = 'results/concatenated_exon_files.csv'
		shell:
				'''
				python3 concat_df.py -i {input.exon_files} -o {output.all_exon_files}
				'''
rule filter_cts:
		input:
				verse = 'results/concatenated_exon_files.csv'
		output:
				filtered = 'results/verse_concat_filtered.csv'
		shell:
				'''
				python filter_cts_mat.py -i {input.verse} -o {output.filtered}
				'''

rule txn_mapping:
		input:
				gtf = 'results/gencode.v45.primary_assembly.annotation.gtf'
		output:
				mapping = 'results/id2gene.txt'
		shell:
				'''
				python parse_gtf.py -i {input.gtf} -o {output.mapping}
				'''